stdlib/arg.ml

```diff
@@ -162,17 +167,17 @@ let parse_and_expand_argv_dynamic_aux allow_expand current argv speclist anonfun
           try
             let keyword, arg = split s in
             assoc3 keyword !speclist, Some arg
-          with Not_found -> stop (Unknown s)
+          with Not_found -> raise (convert_error (Unknown s))
       in
       let no_arg () =
         match follow with
         | None -> ()
-        | Some arg -> stop (Wrong (s, arg, "no argument")) in
+        | Some arg -> raise (Stop (Wrong (s, arg, "no argument"))) in
```

Comment:
When reading the patch (without having tried to understand the code first) I see no particular reason why this `raise` site would not use `convert_error`, while the one above does. I think that this code could be restructured to be easier to understand. In particular, I would like all raises that do (respectively, don't) need the `convert_error` wrapping should be close together, instead of both categories being interleaved with no clear scoping. It would be even better, ideally, if the types could make it more clear that double-handling is impossible (for example by having the *inputs* of `convert_error` use a different error-description type than its output).